File: Aman/Tree/Symmetric Tree.cpp
class Solution {
public:
    bool check(TreeNode* l, TreeNode* r)
    {
        if(l==NULL && r==NULL)
            return true;
        if(l==NULL || r==NULL)
            return false;
        if(l->val!=r->val)
            return false;
        return check(l->left,r->right) && check(r->left,l->right);
    }
    
    
    bool isSymmetric(TreeNode* root) {
        if(root==NULL)
            return true;
        return check(root->left,root->right);
    }
};

File: Harish Kumar/Trees/Symmetric Tree/Symmetric Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean checkTree(TreeNode leftSubTree, TreeNode rightSubTree)
    {
        if(leftSubTree == null && rightSubTree == null)
        {
            return true;
        }
        else if( leftSubTree != null && rightSubTree != null)
        {
            if(leftSubTree.val == rightSubTree.val)
            {
                return checkTree(leftSubTree.left, rightSubTree.right) && checkTree(rightSubTree.left, leftSubTree.right);
            }
        }
        return false;
    }
    public boolean isSymmetric(TreeNode root) {
        if(root == null)
        {
            return true;
        }
        
        return checkTree(root.left, root.right);
        
        
    }
}
